src/pages/index.tsx

```diff
@@ -69,8 +71,31 @@ function IndexPage() {
     },
     validate,
     onSubmit: (values) => {
-      console.log(values);
-      setFormSubmitted(true);
+      fetch("http://139.59.154.199:49160/api/v1/leads", {
+        method: "POST",
+        headers: {
+          Authorization: "secret_token",
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          name: values.username,
+          email: values.email,
+          consentsAccepted: values.acceptance,
+        }),
+      })
+        .then((response) => {
+          if (response.ok) {
+            setFormSubmitted(true);
+          } else if (response.status >= 500 && response.status < 600) {
+            setError5xx(true);
+          } else {
+            setErrorMessage(true);
+          }
+        })
+        .catch((error) => {
+          console.error("Error:", error);
+          setErrorMessage(true);
```

Comment:
**issue:** You haven't implemented setting error message like I explained in my previous PR. You should set string here, not boolaean